src/server/hook_func.c

```diff
@@ -1153,39 +1153,34 @@ mgr_hook_import(struct batch_request *preq)
 	memset(hook_msg, '\0', HOOK_MSG_SIZE);
 	memset(input_path, '\0', MAXPATHLEN+1);
 	memset(output_path, '\0', MAXPATHLEN+1);
-	hookname = preq->rq_ind.rq_manager.rq_objname;
+
+	strncpy(hookname, preq->rq_ind.rq_manager.rq_objname, PBS_MAXSVRJOBID);
 
 	phook = find_hook(hookname);
 
 	if ((phook == (hook *)0) || phook->pending_delete) {
-		snprintf(hook_msg, HOOK_MSG_SIZE-1,
-			"%s does not exist!", hookname);
+		snprintf(hook_msg, HOOK_MSG_SIZE-1, "%s does not exist!", hookname);
```

Comment:
No need for the -1.  snprintf() will do that for you(and with the other snprintf() calls).  Also, can you remove the '!'.  I dislike them in log messages.

Reply:
Regarding '!':
Removal of '!' may turn in to design change as I have seen tests written around them... as this message is used in reply_text which comes to client (aka qmgr command)...
Are you still want me to change?

Reply:
Dang.  If it requires changes to tests then don't bother.  It isn't worth it.  Thanks for looking into it.